sql/moz-fx-data-shared-prod/telemetry_derived/feature_usage_v2/script.sql

```diff
@@ -0,0 +1,760 @@
+-- This script generates the feature usage data for a specific submission_date.
+-- Coercions of values is done automatically based on the type and min_version provided
+-- in the temporary `version_metadata` table. In particular, all nulls for numerical measures
+-- are coalesced to 0 in cases where the client was on an eligible browser version (i.e. a
+-- version where the underlying telemetry was implemented); otherwise leave the measure as null
+-- for ineligible versions.
+-- For adding a new metric:
+-- (1) Add metadata to `version_metadata` for the new metric. Metadata is specified as a list of
+--   STRUCTs where `metric` specifies the column name for the new feature metric in the result table,
+--   `min_version` specifies the minimum eligible browser version and `type` the data type of the
+--   computed metric value.
+-- (2) Update `feature_usages` and specify the feature metric
+-- (3) Update the schema.yaml file and add the new field to the schema
+-- (4) Run `./bqetl query schema deploy telemetry_derived.feature_usage_v2 --force` to update the destination
+--   table schema.
+CREATE TEMP TABLE
+  -- This temporary table contains information about metric types and
+  -- eligible browser versions.
+  version_metadata AS (
+    WITH versions AS (
+      SELECT
+        [
+          STRUCT(
+            "pdf_viewer_time_to_view_ms_content" AS metric,
+            "38" AS min_version,
+            "INT64" AS `type`
+          ),
+          ("pdf_viewer_time_to_view_ms_content_count", "38", "INT64"),
+          ("pdf_viewer_doc_size_kb", "38", "INT64"),
+          ("pdf_viewer_doc_size_kb_content", "38", "INT64"),
```

Comment:
How do we want to store the metadata (eligible browser versions, whether metrics are vetted)? I can think of a couple of options:

1. As common table expression (as it's implemented here). This would have the advantage of having this metadata live right beside the actual query. So if fields are to be added, fewer places need to be updated. The disadvantage is that information about which metrics have been vetted is not really accessible. We could add another `vetted` field here, but this wouldn't be accessible by other tools like Looker.
     * We could add `vetted` metadata to the field description. It would be nice if BigQuery had the option to add tags to columns, there are policy tags, but those aren't really intended for this use case.
2. Store this metadata in a separate table (e.g. `telemetry_derived.feature_usage_meta_v1`). This would make the metadata more accessible and the script here would pretty much work the same way (just need to remove the `version_metadata` temp table). The disadvantage I see is when adding new fields, we'd need to manually update that metadata table. I'm also not sure how Looker would be able to use the vetted information.
3. Track metadata in Google Sheets and connect sheet to BigQuery. Similar to option 2, we can connect a Google sheet to a BigQuery table that can be queried. This might make adding new metadata a little more convenient but still requires changes in multiple places.

Reply:
The immediate use case I had in mind for the `vetted` column was to indicate which metrics should and should not be visible in looker. 

In looker, there is the `hidden` field for measures that controls which can be seen, however I just did some looking around and it doesn't appear that this field can be set contingently (e.g. via a liquid variable). This means that for now (unless we build the look ml files with a script or something) a person will have to manually reference this field to know whether a particular measure should be exposed. Which means that option 1 will work for now.

If we BQ need a table with the metadata, maybe we can stick with this struct as a source of truth but write a small script that scrapes from it and updates the BQ table. That way the metadata only has to be changed in one place (this struct) but can still be accessed via BQ (naive suggestion, don't know how workable this really is).